core/node/rules/can_create_stream.go

```diff
@@ -689,6 +678,22 @@ func (ru *csParams) getNewUserStreamChainAuth() (*auth.ChainAuthArgs, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	// Check for dmPartnerAddress in metadata - this allows users to create their user stream
+	// when they want to DM a bot. The dmPartner must be a bot and the bot module must be installed.
+	if dmPartnerBytes, ok := ru.requestMetadata["dmPartnerAddress"]; ok {
```

Comment:
This looks right, but we aren't adding "dmPartnerAddress" anywhere on the client yet. This can get added later, just calling it out.